iothub/service/src/DirectMethod/Models/DirectMethodRequest.cs

```diff
@@ -57,41 +67,42 @@ public DirectMethodRequest()
         public TimeSpan? ResponseTimeout { get; set; }
 
         /// <summary>
-        /// Get the serialized JSON payload. May be null or empty.
+        /// Get the payload object. May be null or empty.
         /// </summary>
         [JsonIgnore]
-        public string Payload
+        public object Payload
         {
-            get => (string)JsonPayload.Value;
+            get => _payload;
 
             set
             {
-                if (value == null)
-                {
-                    Payload = null;
-                }
-                else
+                _payload = value;
+                if (value != null)
                 {
-                    try
-                    {
-                        JToken.Parse(value);
-                        JsonPayload = new JRaw(value);
-                    }
-                    catch (JsonException ex)
-                    {
-                        throw new ArgumentException(ex.Message, nameof(value));
-                    }
+                    PayloadAsJsonString = JsonConvert.SerializeObject(value);
+                    JsonPayload = new JRaw(PayloadAsJsonString);
                 }
             }
         }
 
+        /// <summary>
+        /// Get the serialized JSON payload. May be null or empty.
+        /// </summary>
+        [JsonIgnore]
+        public string PayloadAsJsonString { get; internal set; }
+
+        /// <summary>
+        /// Get the JSON payload in JRaw type.
+        /// </summary>
+        [JsonProperty("payload")]
+        public JRaw JsonPayload { get; internal set; }
```

Comment:
If possible, I'd consider making this internal. If we ever allow users to pass in their own JSON serailization library, then we'd have trouble constructing and exposing this Newtonsoft type